stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java

```diff
@@ -122,6 +129,34 @@ public void process(K key, V1 value) {
             }
         }
 
+        private boolean joinable() {
+            Set<TopicPartition> partitions = ((ProcessorContextImpl) this.context).task().partitions();
```

Comment:
Didn't this break a KStream test? KStream tests use MockProcessorContext. 
Why did you move joinable() from ProcessorContext to KStreamJoin?